porch/apiserver/pkg/registry/porch/fieldselector.go

```diff
@@ -0,0 +1,103 @@
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+// convertPackageRevisionFieldSelector is the schema conversion function for normalizing the the FieldSelector for PackageRevision
+func convertPackageRevisionFieldSelector(label, value string) (internalLabel, internalValue string, err error) {
+	switch label {
+	case "metadata.name":
+		return label, value, nil
+	case "metadata.namespace":
+		return label, value, nil
+	case "spec.revision", "spec.packageName", "spec.repository":
+		return label, value, nil
+	default:
+		return "", "", fmt.Errorf("%q is not a known field selector", label)
+	}
+}
+
+// convertPackageRevisionResourcesFieldSelector is the schema conversion function for normalizing the the FieldSelector for PackageRevisionResources
+func convertPackageRevisionResourcesFieldSelector(label, value string) (internalLabel, internalValue string, err error) {
+	return convertPackageRevisionFieldSelector(label, value)
+}
+
+// packageFilter filters packages, extending repository.ListPackageRevisionFilter
+type packageFilter struct {
+	repository.ListPackageRevisionFilter
+
+	// Namespace filters by the namespace of the objects
+	Namespace string
+
+	// Repository restricts to repositories with the given name.
+	Repository string
+}
+
+// parsePackageRevisionFieldSelector parses client-provided fields.Selector into a packageFilter
+func parsePackageRevisionFieldSelector(fieldSelector fields.Selector) (packageFilter, error) {
+	var filter packageFilter
+
+	if fieldSelector == nil {
+		return filter, nil
+	}
+
+	requirements := fieldSelector.Requirements()
+	for _, requirement := range requirements {
+
+		switch requirement.Operator {
+		case selection.Equals, selection.DoesNotExist:
+			if requirement.Value == "" {
+				return filter, apierrors.NewBadRequest(fmt.Sprintf("unsupported fieldSelector value %q for field %q with operator %q", requirement.Value, requirement.Field, requirement.Operator))
+			}
+		default:
+			return filter, apierrors.NewBadRequest(fmt.Sprintf("unsupported fieldSelector operator %q for field %q", requirement.Operator, requirement.Field))
+		}
+
+		switch requirement.Field {
+		case "metadata.name":
+			filter.KubeObjectName = requirement.Value
+
+		case "metadata.namespace":
+			filter.Namespace = requirement.Value
+
+		case "spec.revision":
+			filter.Revision = requirement.Value
+		case "spec.packageName":
+			filter.Package = requirement.Value
+		case "spec.repository":
+			filter.Repository = requirement.Value
+
+		default:
+			return filter, apierrors.NewBadRequest(fmt.Sprintf("unknown fieldSelector field %q", requirement.Field))
+		}
+	}
+
+	return filter, nil
+}
+
+// parsePackageRevisionResourcesFieldSelector parses client-provided fields.Selector into a packageFilter
+func parsePackageRevisionResourcesFieldSelector(fieldSelector fields.Selector) (packageFilter, error) {
+	// TOOD: This is a little weird, because we don't have the same fields on PackageRevisionResources.
+	// But we probably should have the key fields
```

Comment:
I was just noticing that; we may want to duplicate at least repo, name, revision

Reply:
Yes, I concur :-)  I can tackle that next!